packages/fuels-programs/src/contract.rs

```diff
@@ -45,6 +84,160 @@ impl Contract {
+            account.add_witnesses(&mut tb)?;
+
+            let tx = tb.build(provider).await?;
+            provider
+                .send_transaction_and_await_commit(tx)
+                .await?
+                .check(None)?;
+        }
+
+        Self::new_loader(&blob_ids, self.salt, self.storage_slots)?
+            .deploy(account, tx_policies)
+            .await
+    }
+
+    /// Splits the contract binary into blobs based on the size specified by `blob_size_policy`.
+    /// This is useful if you prefer to manually deploy the blobs. Once uploaded, you can use [`Contract::new_loader`] to create a loader contract.
+    pub async fn generate_blobs(
+        &self,
+        provider: &Provider,
+        policy: BlobSizePolicy,
+    ) -> Result<Vec<Blob>> {
+        let blob_size = policy.resolve_size(provider).await?;
+
+        let blobs = self
+            .binary
+            .chunks(blob_size)
+            .map(|chunk| Blob::new(chunk.to_vec()))
+            .collect();
+
+        Ok(blobs)
+    }
+
+    // This function creates a contract that loads the specified blobs into memory and delegates the call to the code contained in the blobs.
+    fn loader_contract(blob_ids: &[[u8; 32]]) -> Result<Vec<u8>> {
+        const BLOB_ID_SIZE: u16 = 32;
+        let get_instructions = |num_of_instructions, num_of_blobs| {
+            // There are main 3 steps:
+            // 1. Calculate the total size of the contract
+            // 2. Allocate and load the contract into memory
+            // 3. Jump into the memory where the contract is loaded
+            // After that the execution continues normally with the loaded contract reading our
+            // prepared fn selector and jumps to the selected contract method.
+            [
+                // 0x12 is going to hold the total size of the contract
+                op::move_(0x12, RegId::ZERO),
+                // find the start of the hardcoded blob ids, which are located after the code ends
+                op::move_(0x10, RegId::IS),
+                // 0x10 to hold the address of the current blob id
+                op::addi(0x10, 0x10, num_of_instructions * Instruction::SIZE as u16),
+                // loop counter
+                op::movi(0x13, num_of_blobs),
+                // LOOP starts here
+                // 0x11 to hold the size of the current blob
+                op::bsiz(0x11, 0x10),
+                // update the total size of the contract
+                op::add(0x12, 0x12, 0x11),
+                // move on to the next blob
+                op::addi(0x10, 0x10, BLOB_ID_SIZE),
+                // decrement the loop counter
+                op::subi(0x13, 0x13, 1),
+                // Jump backwards 3 instructions if the counter has not reached 0
+                op::jnzb(0x13, RegId::ZERO, 3),
+                // move the stack pointer by the contract size since we need to write the contract on the stack since only that memory can be executed
```

Comment:
```suggestion
                // 2. Allocate and load the contract into memory
                // move the stack pointer by the contract size since we need to write the contract on the stack since only that memory can be executed
```